content/docs/grammar-language.md

```diff
@@ -170,82 +76,68 @@ Person returns OtherType:
+    'person' name=ID address=Address;
 ```
-Now the ambiguity is resolved and the parser is able to differentiate between the two parser rules.
+In this example, the rule `Person` must start with the `person` keyword followed by an `ID` token and an instance of the `Address` rule.
 
-Parser rules can have many keywords:
+#### Alternatives
+It is possible to match one of multiple valid options by using the pipe operator `|`. The already mentioned `Model` example specifies to parse either `Person` or `Greeting` inside the parentheses:
 ```
-Person:
-    'person' name=ID 'age' age=INT;
+entry Model:
+    (persons+=Person | greetings+=Greeting)*;
 ```
 
+#### Keywords
+Keywords are *inline terminals* which need to match a character sequence surrounded by single or double quotes, for example `'person'` or `"person"`. Keywords must not be empty and must not contain white space.
+
 #### Assignments
+Assignments define properties on the type returned by the surrounding rule.
 There are three different ways to assign an expression (right side) to a property (left side).
 
-1. `=` is used for assigning **only one element** to a property
+1. `=` is used for assigning **a single value** to a property.
     ```
     Person:
         'person' name=ID
     ```
     Here, the property `name` will accept only one expression matching the terminal rule `ID`.
 
-2. `+=` is used to assign **multiple expressions** to a list property
+2. `+=` is used to assign **multiple values** to an array property.
     ```
     Contact:
         addresses+=STRING addresses+=STRING;
     ```
-    Here, the list property `addresses` will accept two expressions matching the terminal rule `STRING`.
-
-    If the assignment has a cardinality of `+` or `*`, then the expressions belong to a single group and must not be interrupted by other expressions.
-    ```
-    Paragraph:
-        'paragraph' (sentences+=STRING)+ id=INT;
-    ```
-    Here, the property `sentences` will accept one or many expressions matching the terminal rule `STRING` followed by an `INT`. The expressions belong to a single group and must not be interrupted by other expressions. The parsing of a document containing:
-    ```
-    paragraph "The expression group " 3 "was interrupted"
-    ```
-    will throw an error since the `STRING` expressions are not continuous. It is however possible to interrupt and resume a sequence of expressions by using [hidden terminal symbols](#hidden-terminal-symbols):
-    ```
-    paragraph "expression one" /* comment */ "expression two" 3
-    ```
-    The above example will be successfully parsed.
+    Here, the array property `addresses` will accept two expressions matching the terminal rule `STRING`.
 
 3. `?=` is used to assign a **value to a property of type boolean**. The value of the property of type `boolean` is set to `true` if the right part of the assignment is consumed by the parser.
     ```
     Employee:
         'employee' name=ID (remote?='remote')?
```

Comment:
Is there a difference between `(remote?='remote')?` and `remote?='remote'?` ?

They both seem to do what I expect in the LSP, is there some differences to be expected in the AST/CST ?

Reply:
No it's exactly the same, so the parentheses are not necessary here.